src/v/datalake/serde_parquet_writer.cc

```diff
@@ -19,11 +19,10 @@ ss::future<writer_error> serde_parquet_writer::add_data_struct(
     auto group = std::get<serde::parquet::group_value>(
       std::move(conversion_result.value()));
     try {
-        co_await _mem_tracker.maybe_reserve_memory(approx_size, as);
         auto stats = co_await _writer.write_row(std::move(group));
         _buffered_bytes = stats.buffered_size;
         _flushed_bytes = stats.flushed_size;
-        _mem_tracker.update_current_memory_usage(_buffered_bytes);
+        co_await _mem_tracker.update_current_memory_usage(_buffered_bytes, as);
```

Comment:
Trying to figure out if there is an upper-bound to the amount of memory we'll temporarily leave-unaccounted for. I think it'd largely be limited to the client-side`max.request.size` (1MiB) or `kafka_batch_max_bytes` (1MiB) for uncompressed messages at least.

Reply:
Thats sounds about right to me, although in practice, that limit worth of data ends up as a single record batch which multiple records in it, right? Here the granularity of update is a record but makes sense to count it as a single record for upper bound estimation.